pkg/operator/ceph/cluster/cluster.go

```diff
@@ -138,80 +137,86 @@ func (c *cluster) detectCephVersion(image string, timeout time.Duration) (*cephv
 }
 
 func (c *cluster) createInstance(rookImage string, cephVersion cephver.CephVersion) error {
-	if c.checkSetOrchestrationRunning() == true {
-		logger.Debugf("As createInstance is currently running added this request as pending.")
-		c.setOrchestrationPending()
-		return nil
-	}
+	var err error
+	c.setOrchestrationNeeded()
 
-	defer c.unsetOrchestrationRunning()
-	initRun := true
-	for c.checkUnsetOrchestrationPending() == true || initRun == true {
-		initRun = false
+	// execute orchestration as long as there is no new orchestration request anymore
```

Comment:
I am interpreting this comment in the opposite sense that it is intended. I think maybe this would be better as "execute orchestration [until] there is no new orchestration request anymore."

This seems like a good place to give an overview of the high-level intention here as well. To my summarization: "execute an orchestration until there are no more unapplied changes to the cluster definition and while no other goroutine is already running a cluster update."
I think this makes it clear in terms of the user's actions what it means when an orchestration is needed and why we need to keep track of whether an orchestration is running.